cpp/modmesh/buffer/pymod/wrap_SimpleArrayPlex.cpp

```diff
@@ -35,21 +35,23 @@ namespace python
 {
 
 /// Execute the callback function with the typed array
+/// @tparam A either `SimpleArrayPlex` or `const SimpleArrayPlex`
 /// @tparam C the type of the callback function
 /// @param arrayplex the plex array, which is the wrapper of the typed array
-/// @param callback the callback function, which has the typed array as the argument
+/// @param callback the callback function, which has the mutable typed array as the argument
 /// @return the return type of the callback function
-template <typename C>
+template <typename A, typename C, typename = std::enable_if_t<std::is_same<A, SimpleArrayPlex>::value || std::is_same<A, const SimpleArrayPlex>::value>>
```

Comment:
Please use `std::remove_const` instead of the duplicated `is_same`: https://en.cppreference.com/w/cpp/types/remove_cv

Reply:
You may also use `is_same_v`, if you want.  C++17 is popular enough and it is ok to depend on it.

Reply:
@yungyuc  fixed.